### ISSUE TITLE
Having problem to attach two entitis (one to many)

### ISSUE THREAD

**Issue by orit1** [TrackableEntities/trackable-entities#233]
Hi we using linq2sql and in the constractor of EntitySet we have option to add 2 Action params for attaching and deatach the entities 
like that: (taken from the designer)

ActivityDates = new EntitySet<ActivityDate>(new Action<ActivityDate>(this.attach_ActivityDates), new Action<ActivityDate>(this.detach_ActivityDates));

how can i perform that in your approach (EF)?

Thanks in advance

**Comment by tonysneed** [comment]
Hi @orit1, I appears your question has to do with EF, rather than Trackable Entities. If you want a hook for setting properties when an entity is saved, the `SavingChanges` event might be what you're looking for: https://coding.abel.nu/2012/12/ef-code-first-dbcontext-savingchanges/. StackOverflow is a good place to post these kinds of questions.

**Comment by weedkiller** [comment]
Hi @tonysneed @orit1 I think the original point would be very useful if trackable entities could help in tracking master-detail or one to many entities easily.

For e.g. this would let us track and audit a Dept with employees, and who was moved. Not necessarily inside EF, but independent of EF / general component or user notifications.

**Comment by tonysneed** [comment]
@weedkiller, Thanks for the follow up on orit1's question. The purpose of Trackable Entities is to take entity state (Added, Updated, Deleted) on entities that are disconnected from a DbContext, and to communicate that state to EF, so that all the changes in an object graph can be saved in a single transaction.

To track what has changed in an object graph (1-M, etc) is a different problem, which Trackable Entities was never designed to solve. However, EF itself has a change tracker on the DbContext that keeps track of all changes in an object graph. The issue, however, with EF6 is that changing the state of an entity (for example, to Added) affects the state of other entities in a graph, and sometimes this can be problematic.

EF Core takes a much cleaner approach when it comes to handling state for disconnected entities, without the adverse behavior in EF6. It also has a recursive `TrackGraph` method that traverses an object graph the way Trackable Entities does. If you are able, I would suggest giving EF Core a try.

**Comment by tonysneed** [comment]
Please re-open is there are more questions about this.

**Comment by weedkiller** [comment]
@tonysneed If EF Core has this, is there a way we can turn an option inside Trackable Entities to track that.

Besides the clean component design for me, using traceable entities, I also believe this can become an Aspect Oriented Design on components not just entities. Since most components have the entities in thier dlls and if we mark them, as trackable it would be easy to do so.. (down the road)  

_______________
_in summary they are just observable method decorators... but just an idea on what yours can do... it has much of it already.._

![image](https://user-images.githubusercontent.com/21086190/69188435-88162880-0aea-11ea-82e9-39155d29cea8.png) [linky..](https://www.postsharp.net/framework)
